go-controller/pkg/node/gateway_localnet.go

```diff
@@ -192,16 +192,23 @@ type localPortWatcherData struct {
 	gatewayIPv4  string
 	gatewayIPv6  string
 	localAddrSet map[string]net.IPNet
+	nodeIP       *net.IPNet
 }
 
-func newLocalPortWatcherData(gatewayIfAddrs []*net.IPNet, recorder record.EventRecorder, localAddrSet map[string]net.IPNet) *localPortWatcherData {
+func newLocalPortWatcherData(gatewayIfAddrs []*net.IPNet, recorder record.EventRecorder, localAddrSet map[string]net.IPNet, nodeIP []*net.IPNet) *localPortWatcherData {
 	gatewayIPv4, gatewayIPv6 := getGatewayFamilyAddrs(gatewayIfAddrs)
-	return &localPortWatcherData{
+
+	lpw := &localPortWatcherData{
 		gatewayIPv4:  gatewayIPv4,
 		gatewayIPv6:  gatewayIPv6,
 		recorder:     recorder,
 		localAddrSet: localAddrSet,
 	}
+
+	if len(nodeIP) > 0 {
+		lpw.nodeIP = nodeIP[0]
```

Comment:
@danwinship I'm a little concerned that this does not handle dual stack. This is the same behavior as shared gw mode. It takes the first ip annotated to the node and uses that for the iptables rules. I think in dual stack we might need to pass all of these and create a rule for each IP. What do you think?